File: packages/shared-types/src/patient.ts
import type { Gender } from "./enums";

export interface Patient {
  id: string;
  userId: string;
  name: string;
  email: string;
  phone: string;
  cpf: string | null;
  birthDate: string | null;
  gender: Gender | null;
  height: number | null;
  weight: number | null;
  goal: string | null;
  medicalNotes: string | null;
  address: string | null;
  emergencyContact: string | null;
  active: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreatePatientDto {
  name: string;
  email: string;
  phone: string;
  cpf?: string;
  birthDate?: string;
  gender?: Gender;
  height?: number;
  weight?: number;
  goal?: string;
  medicalNotes?: string;
  address?: string;
  emergencyContact?: string;
}

export interface UpdatePatientDto extends Partial<CreatePatientDto> {
  active?: boolean;
}

File: apps/backend/src/domain/entities/patient.py
from dataclasses import dataclass, field
from datetime import datetime
import uuid

from src.domain.value_objects.enums import Gender


@dataclass
class Patient:
    id: str
    user_id: str  # Nutritionist owner
    uuid: str = field(
        default_factory=lambda: str(uuid.uuid4())
    )  # Sync key for mobile app
    name: str
    email: str
    phone: str
    cpf: str | None = None
    birth_date: datetime | None = None
    gender: Gender | None = None
    height: float | None = None
    weight: float | None = None
    goal: str | None = None
    medical_notes: str | None = None
    address: str | None = None
    emergency_contact: str | None = None
    active: bool = True
    created_at: datetime = field(default_factory=datetime.utcnow)
    updated_at: datetime = field(default_factory=datetime.utcnow)
